library/block_cipher.c

```diff
@@ -70,6 +123,26 @@ int mbedtls_block_cipher_setkey(mbedtls_block_cipher_context_t *ctx,
                                 const unsigned char *key,
                                 unsigned key_bitlen)
 {
+#if defined(MBEDTLS_BLOCK_CIPHER_SOME_PSA)
+    if (ctx->engine == MBEDTLS_BLOCK_CIPHER_ENGINE_PSA) {
+        psa_key_attributes_t key_attr = PSA_KEY_ATTRIBUTES_INIT;
+        psa_status_t status;
+
+        psa_set_key_type(&key_attr, psa_key_type_from_block_cipher_id(ctx->id));
+        psa_set_key_bits(&key_attr, key_bitlen);
+        psa_set_key_algorithm(&key_attr, PSA_ALG_ECB_NO_PADDING);
+        psa_set_key_usage_flags(&key_attr, PSA_KEY_USAGE_ENCRYPT);
+
+        status = psa_import_key(&key_attr, key, PSA_BITS_TO_BYTES(key_bitlen), &ctx->psa_key_id);
```

Comment:
Not really an issue, but the commit that changes this shows it being previous just `key_bitlen/8`, which would return one short in the case that the key bitlen is not a multiple of 8. Probably hasn't been an issue, as it seems most common for the bitlens to be 8 bits. Regardless, this fixes is, hence this just being a comment.